cmd/fetchd/cmd/stakereconciliationmigrate.go

```diff
@@ -0,0 +1,305 @@
+				migratedEthAddrs   = make(map[AddrString]struct{})
+			)
+			for {
+				row, err = stakesReader.Read()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return fmt.Errorf("failed to read stakes CSV file %s: %w", flagStakesCSVPath, err)
+				}
+
+				ethAddrString = normalizeEthAddr(row[0])
+				if _, ok := migratedEthAddrs[ethAddrString]; ok {
+					// Skip row if ethereum address has been successfully process previously this invocation.
+					continue
+				}
+
+				oldFetchAddrString = row[2]
+				oldFetchAddr, err = sdk.AccAddressFromBech32(oldFetchAddrString)
+				if err != nil {
+					return fmt.Errorf("failed to parse bech32 address for old account %q: %w", oldFetchAddrString, err)
+				}
+
+				registration, ok := registrations.EthAddrMap()[ethAddrString]
+				if !ok {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("no registration found for old account address %q", ethAddrString),
+					})
+					continue
+				}
+
+				newFetchAddr, err := sdk.AccAddressFromBech32(registration.NativeAddress)
+				if err != nil {
+					return fmt.Errorf("failed to parse bech32 address for new account %q: %w", registration.NativeAddress, err)
+				}
+
+				var oldAcctBalance, newAcctBalance sdk.Coins
+				var oldBalanceIndex, newBalanceIndex int
+				for i, b := range bankGenesis.Balances {
+					if b.GetAddress().Equals(oldFetchAddr) {
+						oldAcctBalance = b.GetCoins()
+						oldBalanceIndex = i
+						if newAcctBalance == nil {
+							continue
+						}
+						break
+					}
+
+					if b.GetAddress().Equals(newFetchAddr) {
+						newAcctBalance = b.GetCoins()
+						newBalanceIndex = i
+						if oldAcctBalance == nil {
+							continue
+						}
+						break
+					}
+				}
+
+				if oldAcctBalance == nil {
+					continue
+				}
+
+				var oldAccount, newAccount authtypes.GenesisAccount
+				for _, _account := range genesisAccounts {
+					if _account.GetAddress().Equals(oldFetchAddr) {
+						oldAccount = _account
+						if newAccount == nil {
+							continue
+						}
+						break
+					}
+
+					if _account.GetAddress().Equals(newFetchAddr) {
+						newAccount = _account
+						if oldAccount == nil {
+							continue
+						}
+						break
+					}
+
+				}
+
+				if oldAccount == nil {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("unable to find old account with address %q", oldFetchAddr.String()),
+					})
+					continue
+				}
+
+				if newAccount == nil {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("unable to find new account with address %q", newFetchAddr.String()),
+					})
+					continue
+				}
+
+				if newAcctBalance == nil {
+					return fmt.Errorf("new account with address %q does not have a balance", newFetchAddr.String())
+				}
```

Comment:
a fun note on this check - when testing I ran into a guy that has staked all his new_address tokens after having registered for reconciliation, making it have an actual balance of 0. But here it pops with an empty `[]sdk.Coin{}` `newAcctBalance` which is still valid. Go empty slice vs nil is a quite nasty bug that often goes through!
(so we can keep it, or remove it, I don't think it change much overall)